src/wp-includes/option.php

```diff
@@ -2144,16 +2144,27 @@ function update_network_option( $network_id, $option, $value ) {
 	 */
 	$value = apply_filters( "pre_update_site_option_{$option}", $value, $old_value, $option, $network_id );
 
+	$is_multisite = is_multisite();
 	/*
 	 * If the new and old values are the same, no need to update.
 	 *
-	 * Unserialized values will be adequate in most cases. If the unserialized
-	 * data differs, the (maybe) serialized data is checked to avoid
-	 * unnecessary database calls for otherwise identical object instances.
+	 * An exception applies when no value is set in the database, i.e. the old value is the default.
+	 * In that case, the new value should always be added as it may be intentional to store it rather than relying on the default.
 	 *
-	 * See https://core.trac.wordpress.org/ticket/44956
+	 * See https://core.trac.wordpress.org/ticket/44956 and https://core.trac.wordpress.org/ticket/22192 and https://core.trac.wordpress.org/ticket/59360
 	 */
-	if ( $value === $old_value || maybe_serialize( $value ) === maybe_serialize( $old_value ) ) {
+	if (
+		$value === $old_value ||
+		(
+			false !== $old_value &&
+			/*
+			 * Multisite uses the `meta_value` database field, which is nullable.
+			 * Don't check for values that are equal to `null`.
+			 */
+			( ! $is_multisite || ( null !== $old_value && null !== $value ) ) &&
+			_is_equal_database_value( $old_value, $value )
+		)
+	) {
```

Comment:
```suggestion
         /** This filter is documented in wp-includes/option.php */
	$default_value = apply_filters( "default_site_option_{$option}", false, $option, $network_id );


	if (
		$value === $old_value ||
		(
			false !== $old_value &&
			/*
			 * Multisite uses the `meta_value` database field, which is nullable.
			 * Don't check for values that are equal to `null`.
			 */
			( ! $is_multisite || ( null !== $old_value && null !== $value ) ) &&
			$old_value !== $default_value &&
			_is_equal_database_value( $old_value, $value )
		)
	) {
```

Don't need to worry about the default value filter like in get_option?

Reply:
It doesn't look like the default value was previously taken into consideration, just `false`.

Reply:
> It doesn't look like the default value was previously taken into consideration, just `false`.

I wonder if this is a mistake.

Reply:
I had the same thought when I first saw that, but felt addressing it in this ticket would go outside the scope - and maybe even introduce an issue.

Reply:
Agree with Colin here.

Reply:
Are you okay with addressing the default value in a separate ticket to keep this one in scope, and in case it needs more investigation/a separate tracking history?

Reply:
We should consider opening a separate ticket to thoroughly investigate and share our thoughts there. I briefly checked through the unit tests, but they did not yield the expected results. Therefore, it definitely requires a detailed investigation to understand why this issue was not caught earlier.

@felixarntz and @joemcgill, I would appreciate hearing your thoughts on this matter.

**Unit test:**
```
/**
 * Tests that update_network_option() adds a non-existent option that uses a filtered default value.
 *
 * @ticket 59360
 *
 * @covers ::update_network_option
 */
public function test_update_network_option_should_add_option_with_filtered_default_value() {
	global $wpdb;

	$option        = 'update_network_option_custom_default';
	$default_value = 'default-value';

	add_filter(
		"default_site_option_{$option}",
		static function () use ( $default_value ) {
			return $default_value;
		}
	);

	/*
	 * For a non existing option with the unfiltered default of false, passing false here wouldn't work.
	 * Because the default is different than false here though, passing false is expected to result in
	 * a database update.
	 */
	$this->assertTrue( update_network_option( null, $option, false ), 'update_network_option() should have returned true.' );
	if ( is_multisite() ) {
			$table       = $wpdb->sitemeta;
			$name_field  = 'meta_key';
			$value_field = 'meta_value';
	} else {
			$table       = $wpdb->options;
			$name_field  = 'option_name';
			$value_field = 'option_value';
	}

	$actual = $wpdb->get_row(
		$wpdb->prepare(
			"SELECT %s FROM %s WHERE %s = 'foo' LIMIT 1",
			$value_field,
			$table,
			$name_field
		)
	);

	$this->assertIsObject( $actual, 'The option was not added to the database.' );
	$this->assertObjectHasProperty( $value_field, $actual, "The '$value_field' property was not included." );
	$this->assertSame( '', $actual->$value_field, 'The new value was not stored in the database.' );
}
```

Reply:
@mukeshpanchal27 Sorry for blowing up scope of this change. Create a ticket for the default value, leave to the ticket the code and I will be happy.